### ISSUE TITLE
Add BRAW Toolbox to Proxy clips within an EVENT issue

### ISSUE THREAD

**Issue by leightv** [latenitefilms/BRAWToolbox#124]
Hi,

Apologies if this the same issue as the other post regarding this functionality. I have my proxy files in the same folder. I drag the event containing the ProRes files into BRAW Toolbox and after okaying the Sandbox popup, I get the following error:

<img width="293" alt="Screenshot 2023-04-27 at 18 22 02" src="https://user-images.githubusercontent.com/128590180/234940712-1c67e6a0-61b8-4bb4-baef-e253da8948aa.png">

Thanks in advance.

Leigh

**Comment by latenitefilms** [comment]
Can you try create a new event, with a single `.mov` proxy in it. Make sure the `.braw` is in the same folder as the `.mov`.

Before dragging this event to the Toolbox, select the `.mov` in the Browser, then click **Clip > Open Clip**, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

**Comment by leightv** [comment]
Hi Chris,

I think I’m doing everything right but will check again. Do you have any video clips of this feature being used? That would be the best way for me to check I have the right workflow.

Leigh


Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>



From: Chris Hocking ***@***.***>
Date: Saturday, 29 April 2023 at 22:32
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Can you try create a new event, with a single .mov proxy in it. Make sure the .braw is in the same folder as the .mov.

Before dragging this event to the Toolbox, select the .mov in the Browser, then click Clip > Open Clip, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

**Comment by latenitefilms** [comment]
Basically, I think it is a bug depending on the type of clip you have. However I THINK the workaround is to:

> Before dragging this event to the Toolbox, select the `.mov` in the Browser, then click **Clip > Open Clip**, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

Let me know if this works in your case, and if it does, I can try come up with a proper fix in a future update.

**Comment by leightv** [comment]
Okay I will give it a go. I do think it would be really helpful for have some videos on the toolbox functions as I cannot find any online at all.

Leigh


Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>



From: Chris Hocking ***@***.***>
Date: Sunday, 30 April 2023 at 09:06
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Basically, I think it is a bug depending on the type of clip you have. However I THINK the workaround is to:

Before dragging this event to the Toolbox, select the .mov in the Browser, then click Clip > Open Clip, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

Let me know if this works in your case, and if it does, I can try come up with a proper fix in a future update.

**Comment by latenitefilms** [comment]
Yes, absolutely agree. The main reason the Toolbox documentation is fairly light, and there's no proper tutorials, as they're still kind of a work-in-progress. Aside from "Convert PROJECT to Resolve Friendly FCPXML", the other Toolbox items were really just designed to be "get out of jail" options, for when you really needed to solve something out-of-the-box, as they're all pretty niche for most users. I will however update the documentation and add some videos, once we solve a few more of the Toolbox quirks.

**Comment by leightv** [comment]
Hi Chris,

I tried that and still can’t get it to work.

I think the issue might be I’m just not using it correctly as I’m not quite sure exactly how it’s supposed to work.

What I’d like it to do is take any projects in the event and replace the .mov clips with the equivalent braws but I’m not sure that’s what it does.

I have tried dragging events with both clips and projects in and get the same ‘No BRAW clips were detected’ error.

Am I using it correctly? Or is the issue a sandbox issue as I’ve not enabled access to any drives as far as I can remember.

Leigh



Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>



From: Chris Hocking ***@***.***>
Date: Sunday, 30 April 2023 at 09:06
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Basically, I think it is a bug depending on the type of clip you have. However I THINK the workaround is to:

Before dragging this event to the Toolbox, select the .mov in the Browser, then click Clip > Open Clip, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

Let me know if this works in your case, and if it does, I can try come up with a proper fix in a future update.

**Comment by leightv** [comment]
Hi Chris,

I tried that after your first message suggesting it. It doesn’t seem to work.

Lost

Sent from my iPhone

On 2 May 2023, at 20:43, Leigh Emmerson ***@***.***> wrote:

﻿
Hi Chris,

I tried that and still can’t get it to work.

I think the issue might be I’m just not using it correctly as I’m not quite sure exactly how it’s supposed to work.

What I’d like it to do is take any projects in the event and replace the .mov clips with the equivalent braws but I’m not sure that’s what it does.

I have tried dragging events with both clips and projects in and get the same ‘No BRAW clips were detected’ error.

Am I using it correctly? Or is the issue a sandbox issue as I’ve not enabled access to any drives as far as I can remember.

Leigh



Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074
***@***.******@***.***>
<http://www.povprods.com/>
<image001.jpg>
          <http://www.voicefairy.com/>
<image002.jpg>



From: Chris Hocking ***@***.***>
Date: Sunday, 30 April 2023 at 09:06
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Basically, I think it is a bug depending on the type of clip you have. However I THINK the workaround is to:

Before dragging this event to the Toolbox, select the .mov in the Browser, then click Clip > Open Clip, and apply a Color Board effect to it (actually doesn't matter what the effect is, just an effect that does nothing) - then try again.

Let me know if this works in your case, and if it does, I can try come up with a proper fix in a future update.

**Comment by latenitefilms** [comment]
Here's an example...

I have a ProRes file and a BRAW file in the same folder:

<img width="289" alt="image" src="https://user-images.githubusercontent.com/22286696/235797590-82b76d8e-24a4-4d44-8ad2-2ddd0aed376a.png">

I've dragged the ProRes file into an event:

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/22286696/235799552-4c7a9132-2ca3-425c-8eaa-65269d2b86cd.png">

This next step ideally shouldn't be necessary, but it seems like a bug. With the clip selected, I select **Clip > Open Clip**.

<img width="291" alt="image" src="https://user-images.githubusercontent.com/22286696/235799778-e3786deb-97e2-48c7-928f-7ab7ba4fc212.png">

I then apply a Color Board effect that does nothing:

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/22286696/235799828-f17950bf-c540-4562-ac24-1dcb7cba0c7b.png">

You can now drag the **🎞 RUSHES** event to the **Add BRAW Toolbox to Proxy Clips within an EVENT** drop zone:

<img width="1314" alt="image" src="https://user-images.githubusercontent.com/22286696/235800301-1175e2df-4eb6-4440-8fb6-3e0d4f67c5c4.png">

You'll get this explanation:

<img width="372" alt="image" src="https://user-images.githubusercontent.com/22286696/235800852-5a0c447f-8db4-4992-9740-15faad765c58.png">

It prompts me to Grant Access to the parent folder which contains the rushes. I click **Grant Access**:

<img width="1690" alt="image" src="https://user-images.githubusercontent.com/22286696/235800963-fbe5caf7-700d-441b-9eda-8052a1b8c9cb.png">

I get a success message:

<img width="982" alt="image" src="https://user-images.githubusercontent.com/22286696/235801011-9680c525-75b2-4181-b83e-bb16506789b2.png">

I now need to drag the green icon in the bottom right corner to my Library:

<img width="1315" alt="image" src="https://user-images.githubusercontent.com/22286696/235801075-86347c26-2244-49df-862d-190ff9562f22.png">

I now have two rushes folders:

<img width="431" alt="image" src="https://user-images.githubusercontent.com/22286696/235801133-f5b45742-dfb4-47a6-a579-62ec1c998d6b.png">

If I now **Open Clip** on the new clip, you'll see the BRAW Toolbox effect has been applied, and I can manipulate the RAW metadata:

<img width="1583" alt="image" src="https://user-images.githubusercontent.com/22286696/235801335-bf02275f-9b3a-45f3-86fd-31b19ef74b92.png">

**Comment by leightv** [comment]
Thanks Chris,

I really appreciate you doing this. I followed your process and sadly still no dice however now I know what it does, it’s not that useful to me anyway. What would be much more useful would be a tool that replaced Proxy clips in a project with the braw ones.

I’ve actually had something written myself that does a similar thing. It actually replaces compound clips in a project with ones with the same name in a ‘key project’ xml. I have just tried it so see if it would work with your clips but sadly it doesn’t. I can share my code if you’re interested.

Moving onto what happened when I tried your current tool:

Here’s my folder
[Graphical user interface, table  Description automatically generated]

I import one clip into a rushes event:

[Graphical user interface  Description automatically generated]

Add a colour wheel effect:
***@***.***

Drag it across:
[Graphical user interface, application  Description automatically generated]
and get:

[Graphical user interface, application, website  Description automatically generated]

I hope this is of use.

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Wednesday, 3 May 2023 at 01:38
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Here's an example...

I have a ProRes file and a BRAW file in the same folder:

[image]<https://user-images.githubusercontent.com/22286696/235797590-82b76d8e-24a4-4d44-8ad2-2ddd0aed376a.png>

I've dragged the ProRes file into an event:

[image]<https://user-images.githubusercontent.com/22286696/235799552-4c7a9132-2ca3-425c-8eaa-65269d2b86cd.png>

This next step ideally shouldn't be necessary, but it seems like a bug. With the clip selected, I select Clip > Open Clip.

[image]<https://user-images.githubusercontent.com/22286696/235799778-e3786deb-97e2-48c7-928f-7ab7ba4fc212.png>

I then apply a Color Board effect that does nothing:

[image]<https://user-images.githubusercontent.com/22286696/235799828-f17950bf-c540-4562-ac24-1dcb7cba0c7b.png>

You can now drag the 🎞 RUSHES event to the Add BRAW Toolbox to Proxy Clips within an EVENT drop zone:

[image]<https://user-images.githubusercontent.com/22286696/235800301-1175e2df-4eb6-4440-8fb6-3e0d4f67c5c4.png>

You'll get this explanation:

[image]<https://user-images.githubusercontent.com/22286696/235800852-5a0c447f-8db4-4992-9740-15faad765c58.png>

It prompts me to Grant Access to the parent folder which contains the rushes. I click Grant Access:

[image]<https://user-images.githubusercontent.com/22286696/235800963-fbe5caf7-700d-441b-9eda-8052a1b8c9cb.png>

I get a success message:

[image]<https://user-images.githubusercontent.com/22286696/235801011-9680c525-75b2-4181-b83e-bb16506789b2.png>

I now need to drag the green icon in the bottom right corner to my Library:

[image]<https://user-images.githubusercontent.com/22286696/235801075-86347c26-2244-49df-862d-190ff9562f22.png>

I now have two rushes folders:

[image]<https://user-images.githubusercontent.com/22286696/235801133-f5b45742-dfb4-47a6-a579-62ec1c998d6b.png>

If I now Open Clip on the new clip, you'll see the BRAW Toolbox effect has been applied, and I can manipulate the RAW metadata:

[image]<https://user-images.githubusercontent.com/22286696/235801335-bf02275f-9b3a-45f3-86fd-31b19ef74b92.png>

**Comment by latenitefilms** [comment]
Unfortunately your email didn't come through correct, as it's replying via GitHub - so I didn't get your screenshots. Try replying on the website, or you can email me direct.

So basically, instead of applying BRAW Toolbox to the original media, you'd prefer it replaced the original media with BRAW Toolbox clips?

**Comment by leightv** [comment]
Hi Chris,

Re-sent the email with screenshots. To answer your question, yes. So my workflow would be:


  *   Use Color Finale Transcoder to create Proxy clips.


  *   I post a proxy library to my client.


  *   They email me an XML with their rough edit project.


  *   I import their project.


  *   I then run their edit through a process that replaces every reference to a proxy BRAW clip with the original toolbox clips.


  *   I can then grade and online with all the info.

The existing app I mention actually processes the XML before I import but I’m guessing you can do it with your drag functionality. This would fill in the missing process to make this a true game changer for me.

BTW I’m already now in love with BRAW and shooting on a BMPCC, with your toolbox. It is amazing!

Leigh



Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Friday, 5 May 2023 at 10:57
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Unfortunately your email didn't come through correct, as it's replying via GitHub - so I didn't get your screenshots. Try replying on the website, or you can email me direct.

So basically, instead of applying BRAW Toolbox to the original media, you'd prefer it replaced the original media with BRAW Toolbox clips?

**Comment by latenitefilms** [comment]
The existing Toolbox should still work for that use-case. The library will still be linking to the proxy files, however they won't actually be used, as the BRAW Toolbox effect "replaces" them.

**Comment by leightv** [comment]
Okay I need to get this working so I can see how it actually functions.

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Friday, 5 May 2023 at 11:23
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

The existing Toolbox should still work for that use-case. The library will still be linking to the proxy files, however they won't actually be used, as the BRAW Toolbox effect "replaces" them.

**Comment by leightv** [comment]
Hi Chris,

From my email with the screenshots, can you tell if am I using it correctly?

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Friday, 5 May 2023 at 11:23
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

The existing Toolbox should still work for that use-case. The library will still be linking to the proxy files, however they won't actually be used, as the BRAW Toolbox effect "replaces" them.

**Comment by leightv** [comment]
Hi Chris,

I managed to get it to work! However I can’t see how the clips in my project (sequence) are automatically replaced. I get a new event we BRAW clips but it looks like I have to manually replace them in the edit.

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Friday, 5 May 2023 at 11:23
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

The existing Toolbox should still work for that use-case. The library will still be linking to the proxy files, however they won't actually be used, as the BRAW Toolbox effect "replaces" them.

**Comment by latenitefilms** [comment]
Is your project/timeline in the same event?

**Comment by leightv** [comment]
Yes and I managed to get it to work however the audio is massively out of sync. I think this almost certainly due to the fact my clips are 50p.

Leigh

Sent from my iPhone

On 5 May 2023, at 22:01, Chris Hocking ***@***.***> wrote:

﻿

Is your project/timeline in the same event?

**Comment by leightv** [comment]
Really keen to get this working. Do you want me to open a new subject on GitHub regarding the issue of sync possibly related to frame rate?

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Friday, 5 May 2023 at 22:01
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Is your project/timeline in the same event?

**Comment by latenitefilms** [comment]
Ah yes, that makes sense, as I think BRAW Toolbox handles off-speed clips differently than Color Finale Transcoder. You can try importing the same clip via both Color Finale Transcoder and BRAW Toolbox to compare. Depending on what's different you might be able to manually wrap inside a Compound Clip or something as a workaround (inside the original Clip).

It's weekend here, so I probably won't be able to look too deep into this until next week.

FWIW - the FCPXML's BRAW Toolbox generates should be saved in here:

/Users/YOUR-USER-NAME/Library/Group Containers/A5HDJTY9X5.com.latenitefilms.BRAWToolbox/Library/Application Support/

So if you're familiar with FCPXML, you can try manually modifying before importing, etc.

I'll prioritise a day next week on this workflow so hopefully we can come up with something more seamless!

**Comment by leightv** [comment]
Thanks Chris,

Really appreciate your time to look at this. As I say, working with BRAW with your software has been game changing for me. I’ve just sold my A7S Mk II because I am so impressed with what I can now do with my BMPCCs!

Technically, I don’t think the clips are off-speed, their base rate is 50 fps. Let me know if you need any examples to work with.

In terms of timescale, next week is more than fine for me. I’m just grateful you’re looking into this!

Leigh

Leigh Emmerson
Digital Media Specialist

6 Newman’s Court
Farnham
Surrey
GU9 0SJ

+44(0)1252 506074<tel:+441252506074>
***@***.******@***.***>
[POV Productions Logo]<http://www.povprods.com/>          [Voice Fairy Logo] <http://www.voicefairy.com/>


From: Chris Hocking ***@***.***>
Date: Saturday, 6 May 2023 at 00:38
To: latenitefilms/BRAWToolbox ***@***.***>
Cc: leightv ***@***.***>, Author ***@***.***>
Subject: Re: [latenitefilms/BRAWToolbox] Add BRAW Toolbox to Proxy clips within an EVENT issue (Issue #124)

Ah yes, that makes sense, as I think BRAW Toolbox handles off-speed clips differently than Color Finale Transcoder. You can try importing the same clip via both Color Finale Transcoder and BRAW Toolbox to compare. Depending on what's different you might be able to manually wrap inside a Compound Clip or something as a workaround (inside the original Clip).

It's weekend here, so I probably won't be able to look too deep into this until next week.

FWIW - the FCPXML's BRAW Toolbox generates should be saved in here:

/Users/YOUR-USER-NAME/Library/Group Containers/A5HDJTY9X5.com.latenitefilms.BRAWToolbox/Library/Application Support/

So if you're familiar with FCPXML, you can try manually modifying before importing, etc.

I'll prioritise a day next week on this workflow so hopefully we can come up with something more seamless!

**Comment by latenitefilms** [comment]
I'm going to close this for now, as it's now solved by the Media Extension.
